### DISCUSSION TITLE
跨服务器连接go-http

### DISCUSSION THREAD

**Discussion by Bearlele** [Tsuk1ko/CQ-picfinder-robot#296] [问题求助] [answered]
请问如何跨服务器连接go-http 两台服务器已经组网 可以互相ping通 我宜家把cq-picsearcher-bot配置里的host改成了go-http所在服务器的IP 但还是提示连接失败

**Comment by Tsuk1ko** [comment]
有没有一种可能，go-cqhttp 所在服务器端口没开

**Comment by Tsuk1ko** [comment, marked as answer]
还有，go-cqhttp 配置也检查下监听 ip，需要把 `127.0.0.1` 改成 `0.0.0.0`
![image](https://user-images.githubusercontent.com/24877906/156145433-0b60cc70-6ce4-4fba-b6bf-65cf668fdbec.png)

**Reply by Bearlele** [reply]
go-cqhttp![image](https://user-images.githubusercontent.com/30120610/156146200-e12f8fc8-1c82-45ed-8571-153980d59400.png)
cq-picsearcher-bot![image](https://user-images.githubusercontent.com/30120610/156146242-36942c7e-c697-49a1-8e38-2be948c063d8.png)
你好，两端配置已经改好了，但是启动cq-picsearcher-bot依然提示
0|cqps     |     at ClientRequest.emit (events.js:400:28)
0|cqps     |     at ClientRequest.emit (domain.js:475:12)
0|cqps     |     at Socket.socketErrorListener (_http_client.js:475:9)
0|cqps     | 3/1/2022, 5:48:21 PM 连接错误[/api]
0|cqps     | SocketError: Failed to establish the websocket connection.
0|cqps     |     at W3CWebSocket._sock.addEventListener.once (/root/cq-picsearcher-bot/node_modules/@tsuk1ko/cq-websocket/src/index.js:458:72)
0|cqps     |     at W3CWebSocket._dispatchEvent [as dispatchEvent] (/root/cq-picsearcher-bot/node_modules/yaeti/lib/EventTarget.js:115:12)
0|cqps     |     at W3CWebSocket.onConnectFailed (/root/cq-picsearcher-bot/node_modules/websocket/lib/W3CWebSocket.js:217:14)
0|cqps     |     at WebSocketClient.<anonymous> (/root/cq-picsearcher-bot/node_modules/websocket/lib/W3CWebSocket.js:59:25)
0|cqps     |     at WebSocketClient.emit (events.js:400:28)
0|cqps     |     at WebSocketClient.emit (domain.js:475:12)
0|cqps     |     at ClientRequest.handleRequestError (/root/cq-picsearcher-bot/node_modules/websocket/lib/WebSocketClient.js:227:14)
0|cqps     |     at ClientRequest.emit (events.js:400:28)
0|cqps     |     at ClientRequest.emit (domain.js:475:12)
0|cqps     |     at Socket.socketErrorListener (_http_client.js:475:9)

**Reply by Tsuk1ko** [reply]
防火墙有检查吗，上面说过 go-cqhttp 所在机器防火墙需要开放 6700/tcp

**Reply by Bearlele** [reply]
检查过了 端口是开放的

**Reply by Bearlele** [reply]
ohh 他莫名其妙可以了 谢谢大佬

**Reply by Tsuk1ko** [reply]
啊这

**Comment by Miuzarte** [comment]
去gocq端的服务器网页后台找防火墙  把gocq的端口号放行tcp，cq那边不用

**Reply by Bearlele** [reply]
已经放行了，使用tcping互相ping端口都通
